scipy/signal/splinemodule.c

```diff
@@ -242,20 +244,29 @@ static PyObject *FIRsepsym2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
   thetype = PyArray_ObjectType(image, NPY_FLOAT);
   thetype = PyArray_MIN(thetype, NPY_CDOUBLE);
   a_image = (PyArrayObject *)PyArray_FromObject(image, thetype, 2, 2);
+  if (a_image == NULL) goto fail;
+
   a_hrow = (PyArrayObject *)PyArray_ContiguousFromObject(hrow, thetype, 1, 1);
+  if (a_hrow == NULL) goto fail;
+
   a_hcol = (PyArrayObject *)PyArray_ContiguousFromObject(hcol, thetype, 1, 1);
-  
-  if ((a_image == NULL) || (a_hrow == NULL) || (a_hcol==NULL)) goto fail;
```

Comment:
Note that merging these `goto fail` conditions together is an anti-pattern because one function might fail then another does `PyErr_Clear()` internally and you end up returning `NULL` with no exception set.